_includes/tutorials/schedule-ktable-ttl-ii/kstreams/code/src/main/java/io/confluent/developer/KafkaStreamsKTableTTLExample.java

```diff
@@ -0,0 +1,209 @@
+    final StoreBuilder<KeyValueStore<String, Long>> storeBuilder = Stores.keyValueStoreBuilder(
+        Stores.persistentKeyValueStore(STATE_STORE_NAME), Serdes.String(), Serdes.Long());
+
+
+
+    builder.addStateStore(storeBuilder);
+
+
+    // Read the input data.
+    final KStream<String, String> stream =
+        builder.stream(inputTopicForStream, Consumed.with(Serdes.String(), Serdes.String()));
+
+    final KStream<String, String> stream2 =
+        builder.stream(inputTopicForTable, Consumed.with(Serdes.String(), Serdes.String()));
+
+    KTable<String, AggregateObject> table = stream2
+        .transform(() -> new TTLKTableTombstoneEmitter<String, String, KeyValue<String, ValueWrapper>>(MAX_AGE,
+            SCAN_FREQUENCY, STATE_STORE_NAME), STATE_STORE_NAME)
+        .groupByKey(Grouped.with(Serdes.String(), new JSONSerde<ValueWrapper>()))
+        .aggregate(AggregateObject::new, (key, value, aggregate) -> {
+          System.out.println("aggregate() - value=" + value);
+          if (value.isDeleted())
+            return null; // signal to tombstone this key in KTable
+          else
+            return aggregate.add((String) value.getValue());
+        }, Materialized.<String, AggregateObject, KeyValueStore<Bytes, byte[]>>as("eventstore")
+            .withKeySerde(Serdes.String()).withValueSerde(new JSONSerde<AggregateObject>()));
+
+
+
+    // Perform the custom join operation.
+    final KStream<String, String> joined = stream.leftJoin(table, (left, right) -> {
+      System.out.println("JOINING left=" + left + " right=" + right);
+      if (right != null) {
+        int size = right.getValues().size();
+        return left + " " + right.getValues().get(size - 1); // concat the last value out of the aggregate
+      }
+      return left;
+    });
+    // Write the join results back to Kafka.
+    joined.to(outputTopic, Produced.with(Serdes.String(), Serdes.String()));
+
+    return builder.build();
+  }
+
+
+
+  public Properties getStreamProps(Properties envProp) {
+    final Properties streamsConfiguration = new Properties();
+    streamsConfiguration.put(StreamsConfig.APPLICATION_ID_CONFIG, envProp.get("application.id"));
+    streamsConfiguration.put(StreamsConfig.BOOTSTRAP_SERVERS_CONFIG,
+        envProp.get("bootstrap.servers"));
+    streamsConfiguration.put(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG, "earliest");
+    // Use a temporary directory for storing state, which will be automatically removed after the
+    // test.
+    streamsConfiguration.put(StreamsConfig.STATE_DIR_CONFIG,
+        TestUtils.tempDirectory().getAbsolutePath());
+    // streamsConfiguration.put(StreamsConfig.MAX_TASK_IDLE_MS_CONFIG, 20000);
+
+
+    // These two settings are only required in this contrived example so that the
+    // streamsConfiguration.put(StreamsConfig.COMMIT_INTERVAL_MS_CONFIG, 0);
+    // streamsConfiguration.put(StreamsConfig.CACHE_MAX_BYTES_BUFFERING_CONFIG, 0);
+    return streamsConfiguration;
```

Comment:
Can we consolidate the properties so that they all come from the properties file?  See `CogroupingStreams.java` for an example.

Reply:
This isn't new though. I can move them but was trying to avoid putting in application properties into KafkaStreams object.

Reply:
> This isn't new though

Yeah, sorry, I should have given you more context.  We're migrating all the Kafka Streams tutorials to be "cloud-ready," and for that, we need to put all the configs in the properties file.  That way, when you start the streams app up, the users only need to point to a config file and pass the `Properties` object around for configuration.  

I'm asking you to do it here because it helps with the conversion process to have a tutorial already using this approach.

Reply:
Done, thank you.